src/System.Data.SqlClient/src/System/Data/SqlClient/SqlTransaction.cs

```diff
@@ -141,47 +160,94 @@ override public void Commit()
 
         protected override void Dispose(bool disposing)
         {
-            if (disposing)
+            Exception e = null;
```

Comment:
How will you track a dispose without explicit rollback/commit? 
What kind of application use case will this benefit?